app/actions/views/channel.js

```diff
@@ -718,22 +721,42 @@ export function loadChannelsForTeam(teamId, skipDispatch = false, isReconnect =
             for (let i = 0; i <= MAX_RETRIES; i++) {
                 try {
                     console.log('Fetching channels attempt', (i + 1), teamId, 'include deleted since', lastConnectAt); //eslint-disable-line no-console
-                    const [channels, channelMembers] = await Promise.all([ //eslint-disable-line no-await-in-loop
+                    const [channels, channelMembers, categories] = await Promise.all([ //eslint-disable-line no-await-in-loop
                         Client4.getMyChannels(teamId, true, lastConnectAt),
                         Client4.getMyChannelMembers(teamId),
+                        Client4.getChannelCategories(currentUserId, teamId),
                     ]);
 
+                    data.categories = categories.categories;
+                    data.categoriesOrder = categories.order;
                     data.channels = channels;
                     data.channelMembers = channelMembers;
                     break;
                 } catch (err) {
+                    // eslint-disable-next-line no-console
+                    console.log(err);
                     if (i === MAX_RETRIES) {
                         const hasChannelsLoaded = state.entities.channels.channelsInTeam[teamId]?.size > 0;
                         return {error: hasChannelsLoaded ? null : err};
                     }
                 }
             }
 
+            if (data.categories) {
+                actions.push({
+                    type: ChannelCategoryTypes.RECEIVED_CATEGORY_ORDER,
+                    data: {
+                        teamId,
+                        order: data.categoriesOrder,
+                    },
+                });
```

Comment:
This is brought over from the webapp - but I think it can be left out as categories are returned back in the correct order; I've left it for now as I assume there will be more work coming down the track around re-ordering, creating / deleting etc.